### PR TITLE
remove percent string support

### PR DESCRIPTION
**Pull request by chlgit**
Maybe it's worth in a separated release, so we can roll back easily.

### CODE REVIEW

**Comment by chlgit** [comment]
After testing, I think one release is fine.

File: tests/config/config_parse_test.py [hunk 1]
```diff
@@ -1157,18 +1131,8 @@ def test_validator_passes_with_context(self):
...
-        template = "The {one} thing I {seven} is {mars}"
-        context = config_utils.ConfigContext(
-            path=None,
-            nodes=None,
-            command_context={'mars': 'ok'},
-            namespace=None,
-        )
-        assert self.validator(template, context)
-
     def test_validator_valid_percent_string(self):
```
Comment (qui) [inline tests/config/config_parse_test.py]: do we still need this test? if we do, let's rename

File: tron/config/config_parse.py [hunk 2]
```diff
@@ -70,14 +70,7 @@ def validator(value, config_context):
...
             valid_context_variable_expr(value, config_context)
-            try:
-                value % context
-                StringFormatter(context).format(value)
-                return value
-            except Exception as e:
-                value % context
-                log.warning("Invalid format string {} at {} {}.\nFall back to % string evaluation".format(e, config_context.path, value))
-                return value
+            return StringFormatter(context).format(value)
```
Comment (qui) [inline tron/config/config_parse.py]: Do the error cases below still apply? i.e. does StringFormatter raise the same exceptions in the same situations?

Comment (chlgit) [reply tron/config/config_parse.py]: I think they are still needed here. For example, KeyError can be for unknown context variables, while TypeError can be not enough argument (Theoretically possible, but might never happen in our use cases).

File: tests/config/config_parse_test.py (hunk 1 above)
Comment (chlgit) [reply tests/config/config_parse_test.py]: I've rewritten the test to see if we don't need a %% escape anymore.

**Pull request merged**